torch/csrc/autograd/init.cpp

```diff
@@ -61,7 +61,11 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
       .def("sequence_nr", &Event::sequence_nr);
 
   m.def("_enable_profiler", enableProfiler);
-  m.def("_disable_profiler", disableProfiler);
+  m.def(
+      "_disable_profiler",
+      disableProfiler,
+      py::arg("cleanup_tls_states") = true,
```

Comment:
are there cases when this argument should be false? i thought that we'd want to clean up local pointer to profiler state anyways

Reply:
From offline discussion:
1) We set this to `false` in child thread since we don't explicitly push those profiler TLS states, so we shouldn't want to pop them in disableProfiler
2) However, there's no issues if we set to `true`, due to the way `ThreadLocalStateGuard` works.